src/compat/db_json.cpp

```diff
@@ -312,6 +373,21 @@ JSON_PRIVATE_ALLOCATOR::Free (void *ptr)
   db_private_free (NULL, ptr);
 }
 
+/*
+* db_json_doc_to_value ()
+* doc (in)
+* value (out)
+* We need this cast in order to use the overloaded methods
+* JSON_DOC is derived from GenericDocument which also extends GenericValue
+* Yet JSON_DOC and JSON_VALUE are two different classes because they are templatized and their type is not known
+* at compile time
+*/
+STATIC_INLINE JSON_VALUE *
+db_json_doc_to_value (const JSON_DOC *doc)
+{
+  return const_cast<JSON_VALUE *> (reinterpret_cast<const JSON_VALUE *> (doc));
```

Comment:
couldn't you just simply have done a reinterpret_cast with JSON_VALUE * ?

Reply:
we need to remove the const because otherwise we get the next error: `E0694	reinterpret_cast cannot cast away const or other type qualifiers`